staging/src/k8s.io/apiserver/pkg/util/proxy/proxy.go

```diff
@@ -117,3 +130,34 @@ func ResolveCluster(services listersv1.ServiceLister, namespace, id string, port
 		return nil, fmt.Errorf("unsupported service type %q", svc.Spec.Type)
 	}
 }
+
+// NewRequestForProxy returns a shallow copy of the original request with a context that may include a timeout for discovery requests
+func NewRequestForProxy(location *url.URL, req *http.Request) (*http.Request, context.CancelFunc) {
```

Comment:
If the peerproxy stuff moves back to kube-aggregator or kube-apiserver, we may not need this relocation